### ISSUE TITLE
Add support for NPM version ranges

### ISSUE THREAD

**Issue by JeevanJames** [JeevanJames/SemverRanges#1]
Refer official NPM docs for available formats: https://docs.npmjs.com/cli/v7/using-npm/semver#ranges

**Comment by justinmchase** [comment]
It would be really nice if the main readme indicated that this was not implemented more clearly.

**Comment by justinmchase** [comment]
I don't think this is just an npm version range either. Go and Rust both appear to use the same version range syntax at least.

https://docs.rs/semver/0.9.0/semver/struct.VersionReq.html
https://github.com/Masterminds/semver#checking-version-constraints

And nearly every library that deals with versions in those languages respectively uses those libraries.
